### ISSUE TITLE
[Elementor Pro] Post Types selected for conditions utilise the wrong intention 

### ISSUE THREAD

**Issue by actual-saurabh** [pojome/elementor#5900]
## Prerequisites
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description
In `elementor-pro/classes/utils.php`, the `get_public_post_types` method uses the `'show_in_nav_menus'` parameter to get the intention of using templates:

```
	public static function get_public_post_types( $args = [] ) {
		$post_type_args = [
			// Default is the value $public.
			'show_in_nav_menus' => true,
		];

		if ( ! empty( $args['post_type'] ) ) {
			$post_type_args['name'] = $args['post_type'];
		}

		$_post_types = get_post_types( $post_type_args, 'objects' );

		$post_types = [];

		foreach ( $_post_types as $post_type => $object ) {
			$post_types[ $post_type ] = $object->label;
		}

		return $post_types;
	}
```

However, as per https://developer.wordpress.org/reference/functions/register_post_type/, 

> 'public' (bool) Whether a post type is intended for use publicly either via the admin interface or by front-end users. While the default settings of `$exclude_from_search`, `$publicly_queryable`, `$show_ui`, and `$show_in_nav_menu`s are inherited from public, each does not rely on this relationship and controls a very specific intention.

> 'show_ui' (bool) Whether to generate and allow a UI for managing this post type in the admin. Default is value of `$public`.

> 'show_in_menu' (bool) Where to show the post type in the admin menu. To work, `$show_ui` must be true. If true, the post type is shown in its own top level menu. If false, no menu is shown. If a string of an existing top level menu (eg. 'tools.php' or 'edit.php?post_type=page'), the post type will be placed as a sub-menu of that. Default is value of `$show_ui`.

Based on these definitions, don't you think one of these parameters is a better indicator of an intention to use templates than the intention of being able to add a post type's post to the navigation menu?

If changed, this will add support for templates for lessons in LifterLMS where lessons don't have an independent existence outside a course and hence `show_in_nav_menus` is obviously false to prevent users from adding them to navigation menus. See: https://github.com/gocodebox/lifterlms/blob/master/includes/class.llms.post-types.php#L308-L342

It'll be really useful for a growing number of Elementor users in the LifterLMS community.

Also see: https://www.facebook.com/groups/lifterlmsvip/permalink/1885455271523177/

## Steps to reproduce

NA

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment
NA

**Comment by dax702** [comment]
Agree with @kmpeter and @actual-saurabh. It's important to be able to create post templates for Lifter LMS lessons with Elementor.

**Comment by markwill890** [comment]
Yes, this hugely valuable for LifterLMS users and agree with the suggestion that using show_in_nav_menus seems a strange choice.

**Comment by SaeedMasri** [comment]
I purchased a third party plugin to edit the lesson post type template with Elementor in LifterLMS installation, Common Elementor!
Please help on this.

**Comment by chrisbadgett** [comment]
There are many LifterLMS <> Elementor users. @actual-saurabh 's fix above would be greatly appreciated.

**Comment by arielk** [comment]
This tweak is now available in Elementor Pro v2.3.0

Thanks for your feedback!

**Comment by Micu22** [comment]
@arielk How to use that hook for the Theme Builder? 
It's already version 2.9.11 and this is still not fully working...
We can edit the lesson post type, but we cannot create a template for lessons using the Theme Builder that is a part of Elementor Pro.
Lessons aren't visible in the dropdown at theme builder conditions window. 
The same problem with assigning different headers/footers to lessons.
